drivers/mmcsd/mmcsd_sdio.c

```diff
@@ -2464,16 +2469,22 @@ static void mmcsd_mediachange(FAR void *arg)
  *
  ****************************************************************************/
 
-static int mmcsd_widebus(FAR struct mmcsd_state_s *priv)
+static int mmcsd_widebus(FAR struct mmcsd_state_s *priv, bool wide)
 {
   int ret;
 
   /* Check if the SD card supports wide bus operation (as reported in the
    * SCR or in the SDIO driver capabililities)
    */
 
-  if ((priv->buswidth & MMCSD_SCR_BUSWIDTH_4BIT) != 0 &&
-      (priv->caps & SDIO_CAPS_1BIT_ONLY) == 0)
+  finfo("Setting BUS width to %s. Card type: %d\n",
```

Comment:
This looks incorrect for several reasons.
1. Removing the SDIO_CAPS_1BIT_ONLY is not expectable. It is a hardware trait. I can have a 4 bit interface and only wire 1 bit. This is set by Kconfig.
2. mmcsd_widebus is changing to wide. The addition of the `wide` looks hack-ish.  
    How can this be better done?